src/graphics/image.js

```diff
@@ -88,25 +89,26 @@ Crafty.c("Image", {
             var self = this;
 
             this.img.onload = function () {
-                self._onImageLoad();
+                self._setupImage(self._drawLayer);
             };
         } else {
-            this._onImageLoad();
+            this._setupImage(this._drawLayer);
         }
 
```

Comment:
`_setupImage` is called from two places: when a layer is attached, and when the image is assigned.  If either step hasn't been completed, it'll simply return.  (The same logic is used for sprites).

I experimented with a few different designs, but this one seemed the most robust.